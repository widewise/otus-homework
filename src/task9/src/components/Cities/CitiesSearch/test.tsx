import React from "react";
import renderer from "react-test-renderer";
import { CitiesSearch } from "./index";

jest.mock("react-router-dom", () => ({
    useNavigate: () => (data: any) => data,
}));

describe(`${CitiesSearch.name}`, () =>{
    //arrange
   const searchString = "city";
   let actual: any;
   describe(`rerender ${CitiesSearch.name}`, () => {
       beforeEach(() =>{
           //act
           actual = renderer.create(<CitiesSearch search={searchString} />).toJSON();
       });

       it("should match snapshot", () =>{
           //assert
           expect(actual).toMatchSnapshot();
       });
   });
});