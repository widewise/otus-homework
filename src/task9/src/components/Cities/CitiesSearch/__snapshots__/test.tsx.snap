// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesSearch rerender CitiesSearch should match snapshot 1`] = `
<div
  className="cities-search"
>
  <label>
    City:
  </label>
  <input
    onChange={[Function]}
    placeholder="Enter city"
    type="text"
    value="city"
  />
</div>
`;
